qiime2/sdk/plugin_manager.py

```diff
@@ -100,44 +127,52 @@ def _integrate_plugin(self, plugin):
 
             self.formats[name] = record
         self.type_formats.extend(plugin.type_formats)
+        self.types[plugin.name] = plugin.types
+
+        for type_format in plugin.type_formats:
+            self._canonical_formats.add(type_format.format)
+            if isinstance(type_format.format,
+                          qiime2.plugin.model.SingleFileDirectoryFormatBase):
+                self._canonical_formats.add(type_format.format.file.format)
```

Comment:
I think this chunk is superfluous, as `self._canonical_formats` isn't very useful, but this is currently being revised, so just don't forget to come back to this.